README.md

```diff
@@ -1,52 +1,88 @@
 # Kik Bot API #
-Use this library to develop bots for [Kik Messenger](https://www.kik.com) that are essentially automated humans.
+The Unofficial Kik Bot API is a Python library developed to automate interactions on [Kik Messenger](https://www.kik.com) that are essentially automated humans.
 
-It basically lets you do the same things as the official Kik app by pretending to be a real smartphone client; It communicates with Kik's servers at `talk1110an.kik.com:5223` over a modified version of the [XMPP](https://xmpp.org/about/technology-overview.html) protocol.
+It's essentially a way to create bots that behave like humans on the platform. This library enables your bot to interact with the official Kik app by emulating a real smartphone client. It communicates with Kik's servers at `talk1110an.kik.com:5223` over a modified version of the [XMPP](https://xmpp.org/about/technology-overview.html) protocol.
+
+This library is ideal for developers, hobbyists, and businesses who want to build automated bots to interact with users, groups, and other bots on Kik.
+
+We do not endorse the use of this library for spamming or other malicious purposes. Please use this library responsibly.
 
-This is the new branch of this project and is recommended.
 ## Installation and dependencies ##
-First, make sure you are using **Python 3.6+**, not python 2.7. Second, just install it directly from GitHub:
+Make sure you have Python 3.8 or above installed on your system. You can install this library directly from GitHub:
 ```
 git clone -b new https://github.com/tomer8007/kik-bot-api-unofficial
 pip3 install ./kik-bot-api-unofficial
 ```
-## Usage ##
-Examples are a great way to understand things. A good place to start is the `examples/` directory. 
+## Quick Start Guide ##
+Here's a simple example of how to use the Kik Bot API:
 
-It is as simple as:
 ```python
 from kik_unofficial.client import KikClient
 from kik_unofficial.callbacks import KikClientCallback
 import kik_unofficial.datatypes.xmpp.chatting as chatting
 
+# Your kik login credentials (username and password)
+username = "your_kik_username"
+password = "your_kik_password"
+
+# The bot class that handles all the callbacks from the kik client
 class EchoBot(KikClientCallback):
     def __init__(self):
-        self.client = KikClient(self, "your_kik_username", "your_kik_password")
+        self.client = KikClient(self, username, password)
         self.client.wait_for_messages()
 
+    # This method is called when the bot is authenticated
     def on_authenticated(self):
         self.client.request_roster() # request list of chat partners
 
+    # This method is called when the bot receives a chat message in your dms
```

Comment:
Do we really need this comment?

Reply:
i mostly put it to differentiate it from group messages, but I could remove it

Reply:
Oh OK, but now you removed the groups method so I think we can remove it. Leaving the comment in `echo_bot.py` is good though